pkg/controller/plan/adapter/openstack/destinationclient.go

```diff
@@ -0,0 +1,92 @@
+package openstack
+
+import (
+	"context"
+	"path"
+
+	"github.com/konveyor/forklift-controller/pkg/apis/forklift/v1beta1"
+	"github.com/konveyor/forklift-controller/pkg/apis/forklift/v1beta1/plan"
+	plancontext "github.com/konveyor/forklift-controller/pkg/controller/plan/context"
+	liberr "github.com/konveyor/forklift-controller/pkg/lib/error"
+	core "k8s.io/api/core/v1"
+	k8serr "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/labels"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	k8sutil "sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+type DestinationClient struct {
+	*plancontext.Context
+}
+
+// Delete OpenstackVolumePopulator CustomResource list.
+func (r *DestinationClient) DeletePopulatorDataSource(vm *plan.VMStatus) error {
+	populatorCrList, err := r.getPopulatorCrList()
+	if err != nil {
+		return err
+	}
+	for _, populatorCr := range populatorCrList.Items {
+		err = r.DeleteObject(&populatorCr, vm, "Deleted OpenstackPopulator CR.", "OpenstackVolumePopulator")
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Set the OpenstackVolumePopulator CustomResource Ownership.
+func (r *DestinationClient) SetPopulatorCrOwnership() (err error) {
+	populatorCrList, err := r.getPopulatorCrList()
+	if err != nil {
+		return
+	}
+	for _, populatorCr := range populatorCrList.Items {
+		pvc := core.PersistentVolumeClaim{}
+		err = r.Destination.Client.Get(context.TODO(), client.ObjectKey{Namespace: r.Plan.Spec.TargetNamespace, Name: populatorCr.Spec.ImageID}, &pvc)
+		if err != nil {
+			continue
+		}
+		err = k8sutil.SetOwnerReference(&pvc, &populatorCr, r.Scheme())
+		if err != nil {
+			continue
+		}
+		err = r.Destination.Client.Update(context.TODO(), &populatorCr)
```

Comment:
shouldn't we patch here as well?

Reply:
even if we don't watch the resource, why overriding the entire configuration if we update a single field?
the classic example of this is when the definition you have at hand doesn't include all properties and then those that are set on the entity which you haven't set, will be overridden